### PR TITLE
ME input bus, gives the multiblock direct access to the 16 selected item types

### PR DESCRIPTION
**Pull request by repo-alt**
image
(Looks like UI design borrowed from GT item filter is a bit questionable, but someone may draw a different texture: 
GT5-Unofficial/src/main/resources/assets/gregtech/textures/gui/InputBusME.png )

This is the first type of an input bus, designed as a replacement to the (interface -> conveyor -> super bus) combo

### CODE REVIEW

**Comment by S4mpsa** [comment]
I retextured the GUI and changed the slots from 16 to 9 to match a normal interface.
image

Also made the lower slots work like an interface (almost), showing a ghost item if the input bus is receiving items and empty if the bus cannot get any more items.

Tested briefly, worked as intended.

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_FusionComputer.java [hunk 1]
```diff
@@ -358,7 +358,7 @@ public void onPostTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
                 mEfficiency = 0;
             if (mRunningOnLoad && checkMachine(aBaseMetaTileEntity, mInventory[1])) {
                 this.mEUStore = aBaseMetaTileEntity.getStoredEU();
-                checkRecipe(mInventory[1]);
```
Comment (S4mpsa) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_FusionComputer.java:361]: Seems like an unrelated change, likely something to do with the new `checkRecipe()` function?

**Review by S4mpsa — APPROVED** [review]
Should let Repo review the GUI change before merging.

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_FusionComputer.java (hunk 1 above)
Comment (repo-alt) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_FusionComputer.java:361]: Well, that seem to be unnecessary now since the fusion reactor does not use input buses, but if the concept of pre/post-recipe processing would be applied to fuid buses as well, that will become useful. So I just change it right now for consistency

**Comment by repo-alt** [comment]
I kinda wanted to avoid borrowing AE GUI, since it is unlike GT, but since this is effectively a crossmod I guess it's fine.
Also I noticed that relying on `updateSlots` is unnecessary now that I added `startRecipeProcessing/endRecipeProcessing` brackets, so, the sync work may be done directly in `endRecipeProcessing`, will check and push that now.
Also I actually didn't check any addon multies, but I think they can be adapted pretty easy

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch.java [hunk 2]
```diff
@@ -179,4 +179,6 @@ public void onPreTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {//in
     }
 
     //To change to other page -> use the setter method -> updateTexture
+
+    public int getCircuitSlot() { return -1; }
```
Comment (miozune) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch.java:183]: can we move this to `MetaTileEntity`? `GT_MetaTileEntity_BasicMachine` has the same named method

Comment (repo-alt) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch.java:183]: idk if the integrated circuit makes any sense for anything except single blocks and input buses/hatches

Comment (miozune) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch.java:183]: We _might_ end up adding ghost circuit slot in multiblock controller or whatever else in the future. Also it removes the need for `GT_Packet_SetConfigurationCircuit_Bus` I guess

Comment (repo-alt) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch.java:183]: Well, it would require some refactoring in single blocks in that case, so, it maybe better left for another PR. I just had to add a circuit slot because the new bus has no "real" slots. The circuit logic is pretty easy to move to the usual input buses (or even MetatileEntity), but it just seems unrelated.

Comment (miozune) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch.java:183]: hmm ok

**Comment by S4mpsa** [comment]
Final design:

image
